### ISSUE TITLE
Puppet functions callable from non-network master

### ISSUE THREAD

**Issue by tomsharratt** [godotengine/godot#42354]
**Godot version:** 3.2.3
**OS/device including version:** Windows 10
**Issue description:** The puppet and puppetsync annotated functions still receive calls from other puppet clients, not just the master.

According to this line: https://github.com/godotengine/godot/blob/00949f0c5fcc6a4f8382a4a97d5591fd9ec380f8/core/io/multiplayer_api.cpp#L94 puppet functions should only be callable from the network master but this is not the case.

**Steps to reproduce:**
Create a puppet function, try call an rpc from the server - this will work.
Try call this same puppet function from a client which isn't the network master - it will still call the puppet function

**Comment by Calinou** [comment]
cc @Calinou

**Comment by Faless** [comment]
@tomsharratt can you provide a minimum reproduction project? i.e. the project with which you generate that output?

**Comment by tomsharratt** [comment]
[PuppetBug.zip](https://github.com/godotengine/godot/files/5287143/PuppetBug.zip)
 Here you go @Faless 

If you start two copies up and have one create_client and the other create_server. Press the button on the server one first and you'll see it work as expected:

Client gets both rpcs called and the server just has the puppetsync called.

However, if you press the button on the client you'll see it doesn't get called on the server (this is correct) but then it does get both puppet and puppetsync called on the client you just sent the rpc from (this isn't correct? you shouldn't be accepting calls where the remote_id is not equal to the network master id?)

**Comment by Faless** [comment]
> However, if you press the button on the client you'll see it doesn't get called on the server (this is correct) but then it does get both puppet and puppetsync called on the client

Yes, they are both called locally, but never accepted from the network (i.e. other connected clients will not call that function).
I can see how this is misleading, but the error here is that you should not call that RPC from the client, and if you do, you will notice it is not called across the network (spitting an error), and thus only the misbehaving client goes out of sync.

We could add extra checks that prints error and skip the call when the client attempts to call a function that it's not allowed to on itself.

**Comment by tomsharratt** [comment]
Totally agree with everything you said. I guess it would just be nice, like you said, to have it spit an error instead of running it locally and having that sync break. I found this issue because I accidentally coded in an RPC call from a client and it would be nice to have something warn me that I'm trying to call an RPC that I shouldn't be.
